### PR TITLE
Framework support for checkerboardOffscreenLayers

### PR DESCRIPTION
**Pull request by jason-simmons**
See https://github.com/flutter/flutter/issues/9473

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/rendering/layer.dart [hunk 1]
```diff
@@ -193,12 +194,17 @@ class PerformanceOverlayLayer extends Layer {
   /// that aid it in making better decisions about caching.
   final bool checkerboardRasterCacheImages;
 
+  /// Whether the compositor should checkerboard layers that are rendered to offscreen
+  /// bitmaps. This can be useful for debugging rendering performance.
+  final bool checkerboardOffscreenLayers;
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/layer.dart]: Can you include a paragraph that discusses how this differs from `checkerboardRasterCacheImages`?

File: packages/flutter/lib/src/widgets/app.dart [hunk 2]
```diff
@@ -91,6 +93,9 @@ class WidgetsApp extends StatefulWidget {
   /// Checkerboards raster cache images.
   final bool checkerboardRasterCacheImages;
 
+  /// Checkerboards layers rendered to offscreen bitmaps.
+  final bool checkerboardOffscreenLayers;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/app.dart]: Please include all the information from the render layer here (also on `checkerboardRasterCacheImages` while we're at it).

File: packages/flutter/lib/src/widgets/performance_overlay.dart [hunk 3]
```diff
@@ -91,11 +93,16 @@ class PerformanceOverlay extends LeafRenderObjectWidget {
   /// that aid it in making better decisions about caching.
   final bool checkerboardRasterCacheImages;
 
+  /// Whether the compositor should checkerboard layers that are rendered to offscreen
+  /// bitmaps. This can be useful for debugging rendering performance.
+  final bool checkerboardOffscreenLayers;
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/performance_overlay.dart]: this class should also have the same documentation as the other classes.

**Comment by Hixie** [comment]
Is this testable?

**Pull request merged**